powertools-cloudformation/src/main/java/software/amazon/lambda/powertools/cloudformation/CloudFormationResponse.java

```diff
@@ -0,0 +1,249 @@
+package software.amazon.lambda.powertools.cloudformation;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.CloudFormationCustomResourceEvent;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import software.amazon.awssdk.http.Header;
+import software.amazon.awssdk.http.HttpExecuteRequest;
+import software.amazon.awssdk.http.HttpExecuteResponse;
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.utils.StringInputStream;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Client for sending responses to AWS CloudFormation custom resources by way of a response URL, which is an Amazon S3
+ * pre-signed URL.
+ * <p>
+ * See https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-lambda-function-code-cfnresponsemodule.html
+ */
+public class CloudFormationResponse {
+
+    /**
+     * Indicates if the function invoking send had successfully completed.
+     */
+    public enum ResponseStatus {
+        SUCCESS, FAILED
+    }
+
+    /**
+     * Representation of the payload to be sent to the event target URL.
+     */
+    @SuppressWarnings("unused")
+    static class ResponseBody {
+        private final String status;
+        private final String reason;
+        private final String physicalResourceId;
+        private final String stackId;
+        private final String requestId;
+        private final String logicalResourceId;
+        private final boolean noEcho;
+        private final Object data;
+
+        ResponseBody(CloudFormationCustomResourceEvent event,
+                     Context context,
+                     ResponseStatus responseStatus,
+                     Object responseData,
+                     String physicalResourceId,
+                     boolean noEcho) {
+            requireNonNull(event, "CloudFormationCustomResourceEvent");
```

Comment:
Can we use the method from Objects class itself ? which has requireNonNull

Reply:
Good call.